### PR TITLE
Domain setting

### LINKED ISSUE #308: Add Domain Setting

### LINKED ISSUE #340: Update premapp to support various premd deployment configurations
## Summary

`premapp` should adapt its method of invoking `premd` (backend) based on different deployment types. These deployment types affect the HTTP request URL and any additional headers that may be required.

## Deployment Types

1. **Localhost**: Both `premapp` & `premd` run locally(dev env) without a proxy.
2. **Cloud-Packaged with Proxy**: Backend running behind a proxy and both `premapp` & `premd` are packaged.
3. **Cloud-Packaged without Proxy**: Backend not running behind a proxy but both are packaged.
4. **Cloud-Unpackaged with Proxy**: Backend running behind a proxy.
5. **Cloud-Unpackaged without Proxy**: Backend not running behind a proxy.

> **Note**: *Packaged* means `premapp` & `premd` are running on the same machine.

### Introduce New Environment Variable

Add a new environment variable `PROXY_ENABLED` with possible combinations as follows:

- **Localhost**: `PROXY_ENABLED=false, VITE_IS_PACKAGED=false, BACKEND_URL=""`
- **Cloud-Packaged with Proxy**: `PROXY_ENABLED=true, VITE_IS_PACKAGED=true, BACKEND_URL=""`
- **Cloud-Packaged without Proxy**: `PROXY_ENABLED=false, VITE_IS_PACKAGED=true, BACKEND_URL=""`
- **Cloud-Unpackaged with Proxy**: `PROXY_ENABLED=true, VITE_IS_PACKAGED=false, BACKEND_URL={PROXY_URL}`
- **Cloud-Unpackaged without Proxy**: `PROXY_ENABLED=false, VITE_IS_PACKAGED=false, BACKEND_URL={PREMD_URL}`

### Conditional HTTP Requests

If the backend is running behind a proxy, two scenarios can happen:

1. **A Record with Wildcard not created**: If an A Record is not created for a domain with a wildcard `*`, add a `Host: premd.docker.localhost` header and send request to {HOST}
2. **Existing A Record**: If an A Record exists, the `Host` header should not be added just send req to {HOST}

To check if a domain is created, invoke `GET dnsd.{HOST}/dns/existing`. 
To find HOST use window.location.host if IS_PACKAGED=true, if not than it needs to come through BACKEND_URL env var.

> **Optimization**: Cache the first successful check for an existing A Record in memory to avoid repeated calls to `dnsd`.

### Method Updates

Update the existing `getBackendUrl` method:

1. Update `backendURL` to point to the correct `premd` port: 
    ```javascript
    let backendURL = "http://localhost:54321";
    ```
2. Remove the path `/api/` in `backendURL`:
    ```javascript
    backendURL = `${window.location.protocol}//${window.location.host}/api/`
    ```

### PR DESCRIPTION
**Pull request by Janaka-Steph**
It closes #308 
It closes #340 

Add UI to enter a domain manually.
Update all requests to add Host header if daemon behind proxy.

### CODE REVIEW

**Comment by tiero** [comment]
We should remove the info in the service detail page about running port as not relevant, and rather have a link to the baseURL now

cc/ @filopedraz 
image

**Comment by Janaka-Steph** [comment]
> We should remove the info in the service detail page about running port as not relevant, and rather have a link to the baseURL now
> 
> cc/ @filopedraz image

Just the line `Running Port       8111`?
Which your screenshot is not showing

File: entrypoint.sh [hunk 1]
```diff
@@ -1,10 +1,11 @@
 #!/bin/sh
-set -eo pipefail
+set -eo
```
Comment (tiero) [inline entrypoint.sh:2]: Reason for removing this?

File: src/shared/store/setting.ts [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+      ((window as any).VITE_BACKEND_URL !== undefined || import.meta.env.VITE_BACKEND_URL) &&
+      ((window as any).VITE_BACKEND_URL !== "" || import.meta.env.VITE_BACKEND_URL !== "")
+    );
+  };
   if (isBackendSet()) {
     backendURL = (window as any).VITE_BACKEND_URL || import.meta.env.VITE_BACKEND_URL;
   }
   if (isPackaged()) {
-    backendURL = `${window.location.protocol}//${window.location.host}/api/`;
+    backendURL = `${window.location.protocol}//${window.location.host}/`;
```
Comment (tiero) [inline src/shared/store/setting.ts:19]: Are we suffixing with / by default?

Comment (Janaka-Steph) [reply src/shared/store/setting.ts:19]: Well, in that line yes

**Comment by tiero** [comment]
Resolve conflicts

File: entrypoint.sh (hunk 1 above)
Comment (Janaka-Steph) [reply entrypoint.sh:2]: In POSIX sh, set option pipefail is undefined https://github.com/koalaman/shellcheck/wiki/SC2039

**Pull request merged**